mcp_server/gitlab_tools.py

```diff
@@ -266,3 +266,89 @@ async def add_blocking_merge_request_comment(
         return f"Successfully added blocking comment to merge request {merge_request_url}"
     except Exception as e:
         raise ToolError(f"Failed to add blocking comment to merge request: {e}")
+
+
+async def retry_job(
+    job_url: Annotated[
```

Comment:
I would rather base this on job ID, provided they are unique across projects.

Reply:
The logic of the function uses the URL for the API calls to retry the job, and it needs to know what project it is (gets it from the URL)

Reply:
My point was we don't need a fragile URL parsing if it's not needed - the caller will know the job ID, as well as project URL.

Reply:
Okay I changed it to the combination as you suggested in the latest commit